### PR TITLE
Initial version of elasticai_api package.

### PR DESCRIPTION
**Pull request by brightcoder01**
- `setup.py` and add it in the build script.
- Move `DistributedOptimizer` from elasticdl main package to elasticdl_sdk package.

Relate with issue #2417

### CODE REVIEW

File: setup_sdk.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from setuptools import find_packages, setup
+
+tensorflow_require_list = ["tensorflow"]
+pytorch_require_list = ["torch"]
+
+setup(
+    name="elasticdl_sdk",
```
Comment (workingloong) [inline setup_sdk.py]: We should rename it.

Comment (brightcoder01) [reply setup_sdk.py]: After discussion, we will name it `elasticai_api`. The wheel name is `elasticai-api`.

File: setup_api.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+setup(
+    name="elasticai-api",
+    version="0.2.0rc3.dev0",
+    description="The model development api for ElasticDL.",
+    long_description="This is the sdk for developing ElasticDL models."
+    " Model developers can make slight update using the api in it based on"
+    " their TensorFlow or Pytorch models. And then the model can be trained"
+    " in the elastic and fault-tolerant way.",
+    long_description_content_type="text/markdown",
+    author="Ant Financial",
```
Comment (skydoorkai) [inline setup_api.py]: Ant Group?

File: setup_api.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+tensorflow_require_list = ["tensorflow"]
+pytorch_require_list = ["torch"]
+
+setup(
+    name="elasticai-api",
+    version="0.2.0rc3.dev0",
+    description="The model development api for ElasticDL.",
+    long_description="This is the sdk for developing ElasticDL models."
+    " Model developers can make slight update using the api in it based on"
```
Comment (skydoorkai) [inline setup_api.py]: Model developers can use these APIs to support elastic and fault-tolerant training for their TensorFlow or Pytorch models.

File: setup_api.py (hunk 2 above)
Comment (brightcoder01) [reply setup_api.py]: Good catch, updated.

File: setup_api.py (hunk 3 above)
Comment (brightcoder01) [reply setup_api.py]: Good idea, updated.

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**